PennMobile/DiningCell.swift

```diff
@@ -0,0 +1,260 @@
+    private static let Padding: CGFloat = 25
+    private static let HeaderHeight: CGFloat = 50
+    
+    private let header: UILabel = {
+        let label = UILabel()
+        label.font = UIFont(name: "HelveticaNeue-Light", size: 20)
+        label.text = "Eat at one of these locations"
+        label.textColor = UIColor(r: 115, g: 115, b: 115)
+        label.backgroundColor = .white
+        return label
+    }()
+    
+    private lazy var collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.minimumLineSpacing = DiningCell.InnerWidth //decreases gap between cells
+        let cv = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        cv.backgroundColor = UIColor(r: 248, g: 248, b: 248)
+        cv.dataSource = self
+        cv.delegate = self
+        cv.allowsSelection = false
+        cv.isScrollEnabled = false
+        cv.contentInset = UIEdgeInsets(top: DiningCell.Padding, left: 0, bottom: DiningCell.Padding, right: 0)
+        cv.register(DiningHallCell.self, forCellWithReuseIdentifier: self.diningCell)
+        return cv
+    }()
+    
+    internal var diningHalls: [DiningHall]! {
+        get {
+            return delegate.getDiningHalls().sorted(by: { (hall1, hall2) -> Bool in
+                let time1 = hall1.timeRemaining
+                if time1 == 0 {
+                    return false
+                }
+                
+                let time2 = hall2.timeRemaining
+                
+                if time2 == 0 {
+                    return true
+                }
+                
+                return time1 < time2
+            })
+        }
+    }
+    
+    var delegate: DiningCellDelegate!
+    
+    internal let diningCell = "diningCell"
+    
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:)")
+    }
+    
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+                
+        addSubview(header)
+        addSubview(collectionView)
+        
+        _ = header.anchor(topAnchor, left: leftAnchor, bottom: topAnchor, right: rightAnchor, topConstant: 0, leftConstant: 16, bottomConstant: -DiningCell.HeaderHeight, rightConstant: 0, widthConstant: 0, heightConstant: DiningCell.HeaderHeight)
+        
+        _ = collectionView.anchorToTop(header.bottomAnchor, left: leftAnchor, bottom: bottomAnchor, right: rightAnchor)
+    }
+    
+    public static func calculateCellHeight(numberOfCells: Int) -> CGFloat {
+        if numberOfCells <= 0 { return 0 }
+        
+        let numberOfCells = CGFloat(numberOfCells)
+        
+        let t1 = 2 * Padding
+        let t2 = HallHeight * numberOfCells
+        let t3 = InnerWidth * (numberOfCells - 1)
+        
+        return t1 + t2 + t3 + HeaderHeight
+    }
+    
+    internal func handleMenuPressed(for diningHall: String) {
+        print(diningHall)
+    }
```

Comment:
Remove this function if this is a debugging function only, or implement its functionality.